torch/_prims_common/wrappers.py

```diff
@@ -130,6 +130,9 @@ def _fn(*args, **kwargs):
 
 # TODO: handle tuples of tensors
 def _maybe_resize_out(out: TensorLikeType, shape):
+    if out.shape == shape:
+        return out
+
```

Comment:
Isn't it weird that the check below, on line 139, is for whether the number of elements in the `out` tensor and the shape match? The error message says they must have the same shape, not the same number of elements. Also, while we're here, shouldn't we update it to use something like `utils.warn`, analogous to `utils.check` but for warnings?

Reply:
I think that, given that we don't use warns all that much, it's fine to leave it as it is for now.

As for the logic, I simplified it a bit (not pushed yet).